From f50b4da852341b5882215d623ce89159a25d0aa6 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:56:21 -0400
Subject: [scsi] qla2xxx: Remove Marker type IOCB logic for ISPFX00

Message-id: <1408391817-6389-34-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87939
O-Subject: [PATCH RHEL 7.1 e-stor 33/69] qla2xxx: Remove Marker type IOCB logic for ISPFX00.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Armen Baloyan <armen.baloyan@qlogic.com>

scsi commit id bfd7334e8bc944aca76dd71fdcfe41ac943c84a8

Signed-off-by: Armen Baloyan <armen.baloyan@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_iocb.c b/drivers/scsi/qla2xxx/qla_iocb.c
index 9dad415..7c75766 100644
--- a/drivers/scsi/qla2xxx/qla_iocb.c
+++ b/drivers/scsi/qla2xxx/qla_iocb.c
@@ -524,7 +524,6 @@ __qla2x00_marker(struct scsi_qla_host *vha, struct req_que *req,
 {
  mrk_entry_t *mrk;
  struct mrk_entry_24xx *mrk24 = NULL;
- struct mrk_entry_fx00 *mrkfx = NULL;
 
  struct qla_hw_data *ha = vha->hw;
  scsi_qla_host_t *base_vha = pci_get_drvdata(ha->pdev);
@@ -541,15 +540,7 @@ __qla2x00_marker(struct scsi_qla_host *vha, struct req_que *req,
  mrk->entry_type = MARKER_TYPE;
  mrk->modifier = type;
  if (type != MK_SYNC_ALL) {
-  if (IS_QLAFX00(ha)) {
-   mrkfx = (struct mrk_entry_fx00 *) mrk;
-   mrkfx->handle = MAKE_HANDLE(req->id, mrkfx->handle);
-   mrkfx->handle_hi = 0;
-   mrkfx->tgt_id = cpu_to_le16(loop_id);
-   mrkfx->lun[1] = LSB(lun);
-   mrkfx->lun[2] = MSB(lun);
-   host_to_fcp_swap(mrkfx->lun, sizeof(mrkfx->lun));
-  } else if (IS_FWI2_CAPABLE(ha)) {
+  if (IS_FWI2_CAPABLE(ha)) {
    mrk24 = (struct mrk_entry_24xx *) mrk;
    mrk24->nport_handle = cpu_to_le16(loop_id);
    mrk24->lun[1] = LSB(lun);
diff --git a/drivers/scsi/qla2xxx/qla_mr.c b/drivers/scsi/qla2xxx/qla_mr.c
index 0b5e510..79dd858 100644
--- a/drivers/scsi/qla2xxx/qla_mr.c
+++ b/drivers/scsi/qla2xxx/qla_mr.c
@@ -2140,7 +2140,6 @@ qlafx00_initialize_adapter(scsi_qla_host_t *vha)
  vha->device_flags = DFLG_NO_CABLE;
  vha->dpc_flags = 0;
  vha->flags.management_server_logged_in = 0;
- vha->marker_needed = 0;
  ha->isp_abort_cnt = 0;
  ha->beacon_blink_led = 0;
 
@@ -3211,17 +3210,6 @@ qlafx00_start_scsi(srb_t *sp)
  /* So we know we haven't pci_map'ed anything yet */
  tot_dsds = 0;
 
- /* Forcing marker needed for now */
- vha->marker_needed = 0;
-
- /* Send marker if required */
- if (vha->marker_needed != 0) {
-  if (qla2x00_marker(vha, req, rsp, 0, 0, MK_SYNC_ALL) !=
-      QLA_SUCCESS)
-   return QLA_FUNCTION_FAILED;
-  vha->marker_needed = 0;
- }
-
  /* Acquire ring specific lock */
  spin_lock_irqsave(&ha->hardware_lock, flags);
 
diff --git a/drivers/scsi/qla2xxx/qla_mr.h b/drivers/scsi/qla2xxx/qla_mr.h
index 5f3ed8b..1e60ca2 100644
--- a/drivers/scsi/qla2xxx/qla_mr.h
+++ b/drivers/scsi/qla2xxx/qla_mr.h
@@ -47,30 +47,6 @@ struct cmd_type_7_fx00 {
  uint32_t dseg_0_len;  /* Data segment 0 length. */
 };
 
-/*
- * ISP queue - marker entry structure definition.
- */
-struct mrk_entry_fx00 {
- uint8_t entry_type;  /* Entry type. */
- uint8_t entry_count;  /* Entry count. */
- uint8_t handle_count;  /* Handle count. */
- uint8_t entry_status;  /* Entry Status. */
-
- uint32_t handle;  /* System handle. */
- uint32_t handle_hi;  /* System handle. */
-
- uint16_t tgt_id;  /* Target ID. */
-
- uint8_t modifier;  /* Modifier (7-0). */
- uint8_t reserved_1;
-
- uint8_t reserved_2[5];
-
- uint8_t lun[8];   /* FCP LUN (BE). */
- uint8_t reserved_3[36];
-};
-
-
 #define STATUS_TYPE_FX00 0x01  /* Status entry. */
 struct sts_entry_fx00 {
  uint8_t entry_type;  /* Entry type. */
-- 
1.7.1